File: docs/snippets/all/archetypes/state_change.py
# Log a `StateChange`.

import dalaran as dl

dl.init("dalaran_example_state_change", spawn=True)

dl.set_time("step", sequence=0)
dl.log("door", dl.StateChange(state="open"))

dl.set_time("step", sequence=1)
dl.log("door", dl.StateChange(state="closed"))

dl.set_time("step", sequence=2)
dl.log("door", dl.StateChange(state="open"))

File: docs/snippets/all/archetypes/state_change.rs
//! Log a `StateChange`

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new("dalaran_example_state_change")
        .spawn()?;

    rec.set_time_sequence("step", 0);
    rec.log("door", &dalaran::StateChange::single("open"))?;

    rec.set_time_sequence("step", 1);
    rec.log("door", &dalaran::StateChange::single("closed"))?;

    rec.set_time_sequence("step", 2);
    rec.log("door", &dalaran::StateChange::single("open"))?;

    Ok(())
}
